docs/new-client.md

```diff
@@ -0,0 +1,76 @@
+# Writing a new editor client
+
+Before writing a new editor client, first check if someone else has managed to
+integrate Metals with your favorite text editor.
+
+* [VS Code](https://github.com/scalameta/metals/blob/master/vscode-extension/src/extension.ts),
+  maintained in this repo
+* [Atom](https://atom.io/packages/ide-scala), maintained by
+  [@laughedelic](https://github.com/laughedelic)
+* [Emacs](https://github.com/rossabaker/lsp-scala), maintained by
+  [@rossabaker](https://github.com/rossabaker)
+* Others, see [#217](https://github.com/scalameta/metals/issues/217)
+
+To integrate with Metals with a new editor client, a couple of things should be
+kept in mind
+
+<!-- TOC depthFrom:2 depthTo:2 -->
+
+* [Working directory](#working-directory)
+* [workspace/didChangeWatchedFiles](#workspacedidchangewatchedfiles)
+* [JVM parameters](#jvm-parameters)
+* [Launching the server](#launching-the-server)
+
+<!-- /TOC -->
+
+## Working directory
+
+The server binary needs to be started in the same directory matching the
+`rootUri` parameter of the `initialize` request. Goto definition and other
+SemanticDB-features will not work if the working directory where the metals
+server is started does not match the root directory of the build.
+
+NOTE. Discarding `rootUri` is not strictly compliant with the LSP, consider
+contributing to [#216](https://github.com/scalameta/metals/issues/216) if you
+are affected by this issue.
+
+## workspace/didChangeWatchedFiles
```

Comment:
I suggest to add backticks here, or better make it less LSPy: "Watching file changes"